spatialos-sdk/src/worker/schema/collections.rs

```diff
@@ -133,7 +142,7 @@ where
         }
     }
 
-    fn index(_object: &SchemaObject, _field: FieldId, _index: usize) -> Self::RustType {
+    fn index(_object: &SchemaObject, _field: FieldId, _index: usize) -> Result<Self::RustType> {
         panic!("List fields cannot be indexed into")
```

Comment:
These could now be errors instead of panics?

Reply:
No, they should still be panics since it's effectively indicative of a bug to call `index` on a collection type. I'll be removing these unnecessary methods in the next PR, too.